### ISSUE TITLE
Incompatibility with Twitch?

### ISSUE THREAD

**Issue by Houkime** [QupZilla/qupzilla#2612]
This happened about two days ago. 
Any stream is now just black without sound.
Some player buttons irresponsive.
Upgraded Arch and qupzilla to no avail.

looks like this (a featured broadcast from the main page):
![gifcast_180218190123](https://user-images.githubusercontent.com/32805122/36354059-926938a2-14c6-11e8-9278-fda8e408c1da.gif)

In Firefox everything is fine, so it's possible that Qupzilla is not implementing some part of HTML5 standard correctly.
Probably some change in Twitch player which borked it all for QupZilla.

**Comment by nowrep** [comment]
Works here (Arch with Qt 5.10.1).

Can you see what is in JavaScript console and/or terminal output with `QUPZILLA_ENABLE_JS_OUTPUT=1 qupzilla`?

**Comment by Houkime** [comment]
```
QupZilla: 0 extensions loaded
libpng warning: iCCP: known incorrect sRGB profile
[W] https://player.twitch.tv/js/player.js:28 setStatsEnabled has been deprecated.[W] https://player.twitch.tv/js/player.js:28 setStatsEnabled has been deprecated.[E] https://imasdk.googleapis.com/js/sdkloader/ima3.js:40 Error while parsing the 'allow' attribute: 'autoplay' is an invalid feature name.[E] userscript:_qupzilla_webchannel:432 Uncaught SecurityError: Blocked a frame with origin "https://imasdk.googleapis.com" from accessing a frame with origin "https://www.twitch.tv". The frame being accessed set "document.domain" to "twitch.tv", but the frame requesting access did not. Both must set "document.domain" to the same value to allow access.[E] userscript:_qupzilla_webchannel:432 Uncaught SecurityError: Blocked a frame with origin "https://cdn.krxd.net" from accessing a frame with origin "https://www.twitch.tv". The frame being accessed set "document.domain" to "twitch.tv", but the frame requesting access did not. Both must set "document.domain" to the same value to allow access.QSqlDatabasePrivate::addDatabase: duplicate connection name '94585675137360', old connection removed.
libpng warning: iCCP: known incorrect sRGB profile
libpng warning: iCCP: known incorrect sRGB profile
```

**Comment by cranes-bill** [comment]
Works fine to me as well. Seems the video is just not loaded. And why you expect the buttons to be responsive, when there is no video nor sound!?
Just try with clean profile and check the stream connection!

**Comment by Houkime** [comment]
Found one thing that works - clips.
Clips are playing ok.
Full videos - no.
Streams - no.

Disabling adblocker doesn't help.
Switching from HTML5 player doesn't work (button is not responsive)

**Comment by Houkime** [comment]
![image](https://user-images.githubusercontent.com/32805122/36354963-d5ce09e4-14d3-11e8-9784-acfc2cc2f075.png)
"Show video Stats" in settings

**Comment by Houkime** [comment]
For comparison, here's what it says in Firefox when there is just no video (stream is offline).
![image](https://user-images.githubusercontent.com/32805122/36355040-dbda3064-14d4-11e8-81d4-e44f4702b855.png)

**Comment by cranes-bill** [comment]
So what, the buttons on the video clips are working...
And please stop comparing with Firefox, it is quite different! You can only do compare with another QtWebEngine browser.

**Comment by cranes-bill** [comment]
Nope, the cache is not overflowed. Just as you told some change was made:
>Probably some change in Twitch player which borked it all for QupZilla.

**So the old source was used from the cache before you flush it.**

Regarding the buttons, see how they are looking on another page where the video is not available:
**QupZilla** - inactive controls
![capture du 2018-02-19 08 10 40](https://user-images.githubusercontent.com/5161278/36365132-ae9e0cde-1550-11e8-8dff-0f2e99a92106.png)
**Firefox**
![capture du 2018-02-19 08 14 26](https://user-images.githubusercontent.com/5161278/36365153-c97fd910-1550-11e8-82ec-c47b20e39b07.png)
On Fuckingfox they are still active. I don't know what can control there, perhaps the wind noise or the Moon light....

**Comment by Houkime** [comment]
Understood about controls.

About source not updating if it is already in cache:

So the possible solution to this kind of a problem would be a comparison of source hashsums, right? 
If hash equals, then the source in storage is up to date. If not - reload.
If you could access hashsum or version number without loading the whole source, 
that wouldn't really hurt performance or anything.

However I'm not sure if providing hashes/versions separately is in industry standards currently.
